pkg/utils/utils.go

```diff
@@ -49,3 +57,24 @@ func ValidateNetworkName(networkName string) *types.Error {
 	}
 	return nil
 }
+
+// ValidateInterfaceName will validate the interface name based on the three rules below
+// 1. The name must not be empty
+// 2. The name must be less than 16 characters
+// 3. The name must not contain / or any whitespace characters
+// ref to https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/utils.c?id=1f420318bda3cc62156e89e1b56d60cc744b48ad#n827
+func ValidateInterfaceName(ifName string) *types.Error {
+	if len(ifName) == 0 {
+		return types.NewError(types.ErrInvalidEnvironmentVariables, "interface name is empty", "")
+	}
+	if len(ifName) > maxInterfaceNameLength {
+		return types.NewError(types.ErrInvalidEnvironmentVariables, "interface name is too long", fmt.Sprintf("interface name should be less than %d characters", maxInterfaceNameLength+1))
+	}
+	for _, r := range bytes.Runes([]byte(ifName)) {
```

Comment:
I don't think we want to convert to runes here - the iproute2 code is totally not utf-aware. It's basically just doing `uint8` comparisons.

Reply:
So you think the original change ```bytes.ContainsAny([]byte(ifName), `/ `)``` is more reasonable? But @jellonek think we should check all whitespace-related characters (like \n, \t ...) I'm a little confused.

Reply:
@squeed but what is your suggestion? to list possible characters only from ascii range? or keep original proposal for only two characters

Reply:
While this is technically slightly different from what `iproute2` does, I think this is probably fine.